drgn/helpers/linux/mm.py

```diff
@@ -1590,3 +1591,17 @@ def get_task_rss_info(prog: Program, task: Object) -> TaskRss:
                 shmemrss += rss_stat.count[MM_SHMEMPAGES].value_()
 
     return TaskRss(filerss, anonrss, shmemrss, swapents)
+
+
+def task_vsize(task: Object) -> int:
+    """
+    Return virtual memory size of the task in bytes.
+
+    :param task: ``struct task_struct *``
+    """
+    prog = task.prog_
+    page_size = prog["PAGE_SIZE"].value_()
+    vmem = 0
+    if task.mm:
+        vmem = task.mm.total_vm.value_() * page_size
+    return vmem
```

Comment:
Read the `mm` once to avoid the extra read round-trip and avoid a potential race where it changes:
```suggestion
    mm = task.mm.read_()
    if not mm:
        return 0
    return mm.total_vm.value_() * page_size
```